lib/evmone/instructions_calls.cpp

```diff
@@ -324,6 +321,32 @@ Result eofcreate(
     if (!check_memory(gas_left, state.memory, input_offset_u256, input_size_u256))
         return {EVMC_OUT_OF_GAS, gas_left};
 
+    bytes_view initcontainer;
+    if constexpr (Op == OP_EOFCREATE)
+    {
+        const auto initcontainer_index = uint8_t{pos[1]};
+        pos += 2;
+        const auto& container = state.original_code;
+        const auto eof_header = read_valid_eof1_header(state.original_code);
+        initcontainer = eof_header.get_container(container, initcontainer_index);
+    }
+    else
+    {
+        pos += 1;
+
+        initcontainer =
+            state.get_tx_initcode_by_hash(intx::be::store<evmc::bytes32>(initcode_hash));
+        if (initcontainer.empty())
```

Comment:
Can you actually provide an empty initcontainer in the transaction?

Reply:
Actually you cannot, we forgot to remove this with the introduction of this rule.

Reply:
Please document this.

Reply:
I would change `get_tx_initcode_by_hash` returning `std::optional`